### ISSUE TITLE
A11y_mssql for VSCode_mssql Extension_Select top 1000 Results window_ScreenReader: Screen reader (NVDA/JAWS) is announcing incorrect column header name and incorrect column number while navigating through table elements(Cell data) under "Results" table using arrow keys.

### ISSUE THREAD

**Issue by prabhudevu** [microsoft/vscode-mssql#17622]
## **Actual:**
Screen reader (NVDA/JAWS) is announcing incorrect column header name and incorrect column number while navigating through table elements(Cell data) under "Results" table using arrow keys. For Example "dbo" element is under "Table Schema" column header which is second column but screen reader (NVDA/JAWS) is announcing "dbo IsNullable and column 15" which is completely incorrect.

### **Note:**

1. Issue is also repro with NVDA Screen reader.

## **Expected:**

Screen reader (NVDA/JAWS) should announce correct column header name and column number and row number while navigating through table elements(Cell data) under "Results" table using arrow keys.

**GitHubTags:**#A11yTCS;#Win11;#DesktopApp;#BM_mssql for VSCode_Win32_Mar2023;#A11ySev2;#Benchmark;#mssql for VSCode;#ScreenReader;#JAWS;#FTP;#WCAG1.3.1;#A11yMAS;#Win32;#Element:Table;#AINotInScope;#NVDA;#STP;

## **Environment Details:**
Application Name: mssql for VSCode
Operating System: Win11[22H2]
OS Build: 22621.1413
Visual studio code insider: 1.77.0-insider (user setup)
Extension Version: v1.17.1
Screen reader: JAWS[2023.2210.29 ILM], NVDA[2022.4]

## **Repro Steps:**

1. Open Visual studio code
2. Tab till extension button and search for "mssql".
3. Tab till install and move to the SQL Server button.
4. Tab till Add Connection Button and insert the valid credentials.
5. Tab till the database and hit enter.
6. Tab till the System database and hit enter.
7. Tab till the Tables and hit enter.
8. Tab till any table under System tables and hit shift + F10
9. Tab till Select Top 1000 and hit enter
10. Tab till Execute Query and hit enter
11. Tab till "Results" dropdown and hit enter
12. Turn on the screen reader JAWS.
13. Tab till elements in the table and navigate using arrow keys 

## **User Impact:**
Screen reader users will get complete wrong information of cell data in results table which will misguide them.

[“Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

## **Attachment**
![17622_A11y_mssql for VSCode_mssql Extension_Select top 1000 Results window_ScreenReader (2)](https://user-images.githubusercontent.com/98821792/225856245-b0ec47af-ec6b-44de-acf5-b841717eb4c5.gif)
![17622_A11y_mssql for VSCode_mssql Extension_Select top 1000 Results window_ScreenReader (1)](https://user-images.githubusercontent.com/98821792/225856641-0a1cb7a0-b70b-447a-9244-8c245921d9d3.gif)

**Comment by alanrenmsft** [comment]
@prabhudevu please use the extension installer from here: https://microsoft-my.sharepoint.com/:u:/p/alanren/EQm7x3UTBmBMpsX5hOywRBAB0-RfVIeErCrSsxKJs8OBBg?e=e10YDN to verify the fix, you can follow the instructions here: https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix

**Comment by prabhudevu** [comment]
Hi @alanrenmsft I have verified the issue in the below version after installing the provided extension

Version: 1.79.0-insider (user setup)
Commit: 2946b1ee55db1d406526d3fd1df49250b2f8322d
Date: 2023-05-18T05:25:02.319Z
Electron: 22.5.2
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621

Issue is fixed. Could you please close this issue as fixed.

![17622_Fixed JAWS](https://github.com/microsoft/vscode-mssql/assets/98821792/3d3cdd3e-f49b-408b-9122-595985626fdc)
![17622_fixed NVDA](https://github.com/microsoft/vscode-mssql/assets/98821792/396f30f3-7d16-49ea-b731-3045952113b7)

**Comment by prabhudevu** [comment]
Closed this issue as fixed

**Comment by fsteffi** [comment]
#closed;
